arlo_server/routes.py

```diff
@@ -1246,6 +1267,30 @@ def logout():
     return redirect(f"{base_url}/v2/logout?{params}")
 
 
+@app.route("/auth/superadmin/start")
+def superadmin_login():
+    redirect_uri = urllib.parse.urljoin(request.host_url, SUPERADMIN_OAUTH_CALLBACK_URL)
+    return auth0_sa.authorize_redirect(redirect_uri=redirect_uri)
+
+
+@app.route(SUPERADMIN_OAUTH_CALLBACK_URL)
+def superadmin_login_callback():
+    auth0_sa.authorize_access_token()
+    resp = auth0_sa.get("userinfo")
+    userinfo = resp.json()
+
+    # we rely on the auth0 auth here, but check against a single approved domain.
```

Comment:
why is this necessary?

Reply:
defense-in-depth, I don't want to mistakenly get to a place where we forget to limit users in Auth0, and then anyone with a gmail address gets superadmin.